### PR TITLE
[test-only][full-ci] improve error logging and allow cleanup scripts to run

### PR DESCRIPTION
**Pull request by ishabaral**
## Description
- Improved error logging: log error response body
- do not fail hooks if saving screenshot fails

### CODE REVIEW

File: test/gui/shared/scripts/helpers/api/oc10.py [hunk 1]
```diff
@@ -50,13 +51,24 @@ def create_user(username, password, displayname, email):
...
     }
     response = request.post(url, body)
-    request.assertHttpStatus(response, 200)
+    if response.status_code != 200:
+        root = ET.fromstring(response.content)
+        message = root.find(".//message")
+        raise Exception(
+            "Creating user '%s' failed with %s : " % (username, response.status_code)
+            + message.text
+        )
```
Comment (saw-jan) [inline test/gui/shared/scripts/helpers/api/oc10.py:60]: or just print the response content?
```suggestion
            "Creating user '%s' failed with %s : \n" % (username, response.status_code)
            + response.content
        )
```

File: test/gui/shared/scripts/helpers/api/oc10.py [hunk 2]
```diff
@@ -50,13 +51,24 @@ def create_user(username, password, displayname, email):
...
     # oc10 does not set display name while creating user,
     # so we need update the user info
     user_url = url_join(get_ocs_url(), "users", username)
     display_name_body = {"key": "displayname", "value": displayname}
     display_name_response = request.put(user_url, display_name_body)
-    request.assertHttpStatus(display_name_response, 200)
+    if display_name_response.status_code != 200:
+        raise Exception(
+            "Updating the user '%s' with displayname '%s', failed with %s\n"
```
Comment (saw-jan) [inline test/gui/shared/scripts/helpers/api/oc10.py:68]:
```suggestion
            "Updating the user '%s' with displayname '%s' failed with %s\n"
```

File: test/gui/shared/scripts/helpers/api/oc10.py [hunk 3]
```diff
@@ -50,13 +50,22 @@ def create_user(username, password, displayname, email):
         "email": email,
     }
     response = request.post(url, body)
-    request.assertHttpStatus(response, 200)
+    if response.status_code != 200:
+        raise Exception(
+            "Creating user '%s' failed with %s :" % (username, response.status_code)
```
Comment (saw-jan) [inline test/gui/shared/scripts/helpers/api/oc10.py:55]:
```suggestion
            "Creating user '%s' failed with %s :\n" % (username, response.status_code)
```

File: test/gui/shared/scripts/helpers/api/oc10.py [hunk 4]
```diff
@@ -50,13 +50,22 @@ def create_user(username, password, displayname, email):
...
+        )
     # oc10 does not set display name while creating user,
     # so we need update the user info
     user_url = url_join(get_ocs_url(), "users", username)
     display_name_body = {"key": "displayname", "value": displayname}
     display_name_response = request.put(user_url, display_name_body)
-    request.assertHttpStatus(display_name_response, 200)
+    if display_name_response.status_code != 200:
+        raise Exception(
+            "Updating the user '%s' with displayname '%s' failed with %s\n"
```
Comment (saw-jan) [inline test/gui/shared/scripts/helpers/api/oc10.py:65]:
```suggestion
            "Updating the user '%s' with displayname '%s' failed with %s:\n"
```

File: test/gui/shared/scripts/helpers/api/oc10.py [hunk 5]
```diff
@@ -50,13 +50,22 @@ def create_user(username, password, displayname, email):
         "email": email,
     }
     response = request.post(url, body)
-    request.assertHttpStatus(response, 200)
+    if response.status_code != 200:
+        raise Exception(
+            "Creating user '%s' failed with %s :\n" % (username, response.status_code)
```
Comment (saw-jan) [inline test/gui/shared/scripts/helpers/api/oc10.py:55]:
```suggestion
            "Creating user '%s' failed with %s:\n" % (username, response.status_code)
```

File: test/gui/shared/scripts/bdd_hooks.py [hunk 6]
```diff
@@ -143,31 +143,22 @@ def scenario_failed():
...
-# runs after every scenario
-# Order: 2
 @OnScenarioEnd
 def hook(context):
     clearWaitedAfterSync()
     closeSocketConnection()
 
     # capture a screenshot if there is error or test failure in the current scenario execution
-    if scenario_failed() and os.getenv("CI") and isLinux():
+    if scenario_failed() and isLinux():
```
Comment (saw-jan) [inline test/gui/shared/scripts/bdd_hooks.py:152]:
```suggestion
    if scenario_failed() and os.getenv("CI") and isLinux():
```

File: test/gui/shared/scripts/helpers/api/oc10.py [hunk 7]
```diff
@@ -20,43 +28,28 @@ def checkSuccessOcsStatus(response):
...
+
     # oc10 does not set display name while creating user,
     # so we need update the user info
-    user_url = url_join(get_ocs_url(), "users", username)
+    user_url = get_provisioning_url("users", username)
     display_name_body = {"key": "displayname", "value": displayname}
     display_name_response = request.put(user_url, display_name_body)
-    request.assertHttpStatus(display_name_response, 200)
+    request.assertHttpStatus(
+        display_name_response, 200, f"Failed to update user '{username}' display name"
```
Comment (PrajwolAmatya) [inline test/gui/shared/scripts/helpers/api/oc10.py:49]:
```suggestion
        display_name_response, 200, f"Failed to update display name of user '{username}'"
```

**Review by PrajwolAmatya — APPROVED** [review]
LGTM

**Pull request merged**